src/rgw/rgw_lc.cc

```diff
@@ -408,29 +436,11 @@ int RGWLC::bucket_lc_process(string& shard_id)
           rgw_obj obj(bucket_info.bucket, key);
           RGWObjectCtx rctx(store);
           if (prefix_iter->second.obj_tags != boost::none) {
-            bufferlist tags_bl;
-            int ret = read_obj_tags(store, bucket_info, obj, rctx, tags_bl);
-            if (ret < 0) {
-              if (ret != -ENODATA)
-                ldout(cct, 5) << "ERROR: read_obj_tags returned r=" << ret << dendl;
+            ret = check_objtags(cct, store, prefix_iter, bucket_info, obj);
+            if (ret > 0)
               continue;
-            }
-            RGWObjTags dest_obj_tags;
-            try {
-              auto iter = tags_bl.begin();
-              dest_obj_tags.decode(iter);
-            } catch (buffer::error& err) {
-               ldout(cct,0) << "ERROR: caught buffer::error, couldn't decode TagSet" << dendl;
-              return -EIO;
-            }
-
-            if (!includes(dest_obj_tags.get_tags().begin(),
-                          dest_obj_tags.get_tags().end(),
-                          prefix_iter->second.obj_tags->get_tags().begin(),
-                          prefix_iter->second.obj_tags->get_tags().end())){
-              ldout(cct, 20) << __func__ << "() skipping obj " << key << " as tags do not match" << dendl;
-              continue;
-            }
+            else if (ret == -EIO)
```

Comment:
why only `-EIO`? maybe any `ret < 0`?

Reply:
It only currently returns EIO, but I agree, it should be <0 for future proofing.